test/integration/crcsuite/collect.go

```diff
@@ -0,0 +1,234 @@
+package crcsuite
+
+import (
+	"archive/zip"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/machine"
+	"github.com/code-ready/machine/libmachine/ssh"
+	"github.com/pkg/errors"
+	"golang.org/x/sync/semaphore"
+)
+
+func runDiagnose(dir string) error {
+	filename := filepath.Join(dir, fmt.Sprintf("diagnose-%d.zip", time.Now().Unix()))
+	absolute, err := filepath.Abs(filename)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Writing diagnostics at %s\n", filename)
+	zipFile, err := os.Create(absolute)
+	if err != nil {
+		return err
+	}
+	defer zipFile.Close()
+
+	writer := zip.NewWriter(zipFile)
+	defer writer.Close()
+
+	var collectors = []Collector{
+		&TreeCollector{
+			Dir:    constants.CrcBaseDir,
+			Target: "tree.txt",
+		},
+		file(constants.ConfigPath),
+		file(constants.LogFilePath),
+		file(constants.DaemonLogFilePath),
+		file(filepath.Join(constants.MachineBaseDir, "machines", constants.DefaultName, "config.json")),
+		file("/etc/hosts"),
+		vmFile("/etc/resolv.conf"),
+		vmFile("/etc/hosts"),
+		vmCommand("journalctl -n 1000 --no-pager", "journalctl.txt"),
+		vmCommand("ifconfig", "ifconfig.txt"),
```

Comment:
I think nowadays this has been superseded by `ip addr`

Reply:
:+1: